### PR TITLE
Fix up related posts organism to account for empty related content

### PR DESCRIPTION
**Pull request by kurtw**
Blocks aren't really rendering with clean classes and the sidefoot rendering template doesn't account for empty results to not print a `div`.

## Changes

- Changed up the templates so they account for missing results

## Testing

- go to any page missing results and inspect the sidefoot content. There shouldn't be any empty `div`s.

## Review

- @kave 
- @richaagarwal 
- @rosskarchner 

## Screenshots
before
screen shot 2016-04-15 at 1 12 52 pm
after
screen shot 2016-04-15 at 1 12 13 pm

### CODE REVIEW

File: cfgov/jinja2/v1/_includes/molecules/related-posts.html [hunk 1]
```diff
@@ -16,40 +16,37 @@
 
    ========================================================================== #}
 
-{% macro render(block, is_half_width=false, hide_header_slug=false) %}
+{% macro render(posts, block, is_half_width=false, hide_header_slug=false) %}
     {% import 'macros/category-icon.html' as category_icon %}
```
Comment (richaagarwal) [inline cfgov/jinja2/v1/_includes/molecules/related-posts.html]: I think you need to update the other places that call `related_posts.render(...)` or set posts to None

Comment (kurtw) [inline cfgov/jinja2/v1/_includes/molecules/related-posts.html]: should only be the one

Comment (kurtw) [inline cfgov/jinja2/v1/_includes/molecules/related-posts.html]: :+1:

File: cfgov/jinja2/v1/_includes/templates/render_block.html [hunk 2]
```diff
@@ -1,3 +1,3 @@
-<div class="block {{ block.block.meta.classname if block.block.meta.classname else '' }}">
+<div class="block {{ block.block.meta.classname if block.block.meta.classname else '' -}}">
```
Comment (kave) [inline cfgov/jinja2/v1/_includes/templates/render_block.html]: what does the `-` do?

Comment (jimmynotjim) [inline cfgov/jinja2/v1/_includes/templates/render_block.html]: Removes extra whitespace that Jinja annoyingly adds

**Comment by kave** [comment]
@kurtw can you include an example page

**Comment by kurtw** [comment]
@kave I can't really find one, but you can easily test it with a page by removing its tags!

**Comment by richaagarwal** [comment]
@kave here's an example: http://localhost:8000/about-us/newsroom/cfpb-monthly-complaint-snapshot-examines-debt-collection-complaints/

@kurtw 👍

**Comment by kave** [comment]
👍

**Comment by richaagarwal** [comment]
👍

**Pull request merged**